tests/post/steps/test_volume.py

```diff
@@ -26,3 +56,54 @@ def check_storage_class(host, name, k8s_apiclient):
         if isinstance(exc, ApiException) and exc.status == 404:
             assert False, 'StorageClass {} not found'.format(name)
         raise
+
+
+@then(parsers.parse("the Volume '{name}' is '{status}'"))
+def check_volume_status(host, name, status, k8s_custom_client):
+    def _check_status():
+        try:
+            volume = k8s_custom_client.get_cluster_custom_object(
+                group="storage.metalk8s.scality.com",
+                version="v1alpha1",
+                plural="volumes",
+                name=name
+            )
+        except (ApiException, HTTPError) as exc:
+            if isinstance(exc, ApiException) and exc.status == 404:
+                assert False, 'Volume {} not found'.format(name)
+            raise
+        try:
+            assert volume['status']['phase'] == status,\
+                'Unexpected status: expected {}, got {}'.format(
+                    status, volume['status']['phase']
+                )
+        except KeyError:
+            assert status == 'Unknown', \
+                'Unexpected status: expected {}, got none'.format(status)
+
+    utils.retry(
+        _check_status, times=30, wait=2,
+        name='checking status of Volume {}'.format(name)
+    )
+
+
+@then(parsers.parse("the PersistentVolume '{name}' has size '{size}'"))
+def check_pv_size(host, name, size, k8s_client):
+    def _check_size():
+        try:
+            pv = k8s_client.read_persistent_volume(name)
+        except (ApiException, HTTPError) as exc:
+            if isinstance(exc, ApiException) and exc.status == 404:
+                assert False, 'PersistentVolume {} not found'.format(name)
+            raise
+        assert pv.spec.capacity['storage'] == size, \
+            'Unexpected PersistentVolume size: expected {}, got {}'.format(
+                size, pv.spec.capacity['storage']
+            )
+
+    utils.retry(
+        _check_size, times=30, wait=2,
```

Comment:
Seeing that PV doesn't trigger long action at creation time, I think we may reduce the retry parameters.

Maybe `times=10, wait=2` is enough.